src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -127,14 +127,17 @@ export const QueryEditorUI: React.FC<Props> = (props) => {
   };
 
   const onSubmit = (query: Query, dateRange?: TimeRange) => {
+    // TODO: this is a hack to get the current query string as query.query in some use cases are empty
+    const currentQuery = inputRef.current?.getModel()?.getValue() ?? query.query;
+    if (!currentQuery || currentQuery.trim() === '') return;
```

Comment:
We should not do this. IF i have a vlaid query and then i clear it. This will prevent me from seeing my results